test2code-admin/src/main/kotlin/com/epam/drill/plugins/test2code/Plugin.kt

```diff
@@ -363,16 +363,20 @@ class Plugin(
             .also { logPoolStats() }
 
         is CoverDataPart -> {
-            message.data.groupBy { it.sessionId }
+            message.data
+                .groupBy { it.sessionId }
                 .forEach { (probeSessionId, data) ->
-                    val sessionId = probeSessionId ?: message.sessionId ?: GLOBAL_SESSION_ID
-                    activeScope.activeSessions[sessionId]?.let {
+                    var sessionId = probeSessionId ?: message.sessionId ?: GLOBAL_SESSION_ID
+                    if (activeScope.activeSessions[sessionId] == null){
+                        sessionId = GLOBAL_SESSION_ID
+                    }
+                    activeScope.activeSessions[sessionId].let{
                         activeScope.addProbes(sessionId) { data }?.run {
                             if (isRealtime) {
                                 activeScope.probesChanged()
                             }
                         }
-                    } ?: logger.debug { "Attempting to add coverage in non-existent active session with id $sessionId" }
```

Comment:
We should think about _always_ keeping global session alive (in current backend implementation).
E.g. we can add check to `finishSession` handler to look if a just-finished session were a global one, and if it was - start a new one.
Hacky - yes, but it might work just fine

Reply:
log message was returned